client/src/controllers/LoaderController.test.js

```diff
@@ -0,0 +1,99 @@
+import { Application } from '@hotwired/stimulus';
+import { LoaderController } from './LoaderController';
+
+describe('LoaderController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+            <button
+                type="submit"
+                class="button button-longrunning"
+                data-controller="w-loader"
+                data-w-loader-active-class="button-longrunning-active"
+                data-w-loader-active-value="Signing in…"
+                data-w-loader-duration-seconds-value="40"
+                data-action="w-loader#activate"
+            >
+                <svg>...</svg>
+                <em data-w-loader-target="label" id="em-el">Sign in</em>
+            </button>
+        `;
+    Application.start().register('w-loader', LoaderController);
+  });
+
+  afterEach(() => {
+    document.body.innerHTML = '';
+    jest.clearAllMocks();
+    jest.clearAllTimers();
+  });
+
+  it('should not execute the rest of the code if the form is not valid', () => {
+    const form = document.createElement('form');
+    const button = document.querySelector('.button-longrunning');
+    let returnCalled = false;
+
+    form.noValidate = false;
+    form.checkValidity = jest.fn().mockReturnValue(false);
+    button.closest = jest.fn().mockReturnValue(form);
+
+    if (
+      form &&
+      form.checkValidity &&
+      !form.noValidate &&
+      !form.checkValidity()
+    ) {
+      returnCalled = true;
+      return;
+    }
+
+    expect(returnCalled).toBe(true);
+  });
+
+  it('should not return if the form is  valid', () => {
+    const form = document.createElement('form');
+    const button = document.querySelector('.button-longrunning');
+    let returnCalled = false;
+
+    form.noValidate = true;
+    form.checkValidity = jest.fn().mockReturnValue(true);
+    button.closest = jest.fn().mockReturnValue(form);
+
+    if (
+      form &&
+      form.checkValidity &&
+      !form.noValidate &&
+      !form.checkValidity()
+    ) {
+      returnCalled = true;
+      return;
+    }
+
+    expect(returnCalled).toBe(false);
+  });
+
+  it('should set timeout', () => {
+    const form = document.createElement('form');
+    const button = document.querySelector('.button-longrunning');
+    jest.useFakeTimers();
```

Comment:
You should only have to call this once at the start of the test file. 

https://jestjs.io/docs/timer-mocks